### PR TITLE
[4.0] webcomponents everywhere, media field

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- Move the media field to web components
- Original functionality
- Provides more freedom to do do something more with this ultra standard Bootstrap field

#### REQUESTING FOR SOME BETTER DESIGN
If any designer comes along this PR and they feel that this part needs some redesign love, please do help and provide some mockups

### Testing Instructions

Edit an article and change the intro and full text images

### Expected result

Same functionality as 3.x

### Actual result
screen shot 2017-08-28 at 11 18 32

screen shot 2017-08-28 at 11 18 58

screen shot 2017-08-28 at 11 18 48

### Documentation Changes Required

NOPE

### CODE REVIEW

File: layouts/joomla/form/field/media.php [hunk 1]
```diff
@@ -85,22 +91,21 @@
...
-	data-preview="<?php echo $showPreview ? 'true' : 'false'; ?>"
-	data-preview-as-tooltip="<?php echo $showAsTooltip ? 'true' : 'false'; ?>"
-	data-preview-container=".field-media-preview"
-	data-preview-width="<?php echo $previewWidth; ?>"
-	data-preview-height="<?php echo $previewHeight; ?>"
+<joomla-field-media class="field-media-wrapper"
+	basepath="<?php echo JUri::root(); ?>"
+	rootfolder="<?php echo ComponentHelper::getParams('com_media')->get('file_path', 'images'); ?>"
+	url="<?php echo $url; ?>"
+	modalcont=".modal"
```
Comment (brianteeman) [inline layouts/joomla/form/field/media.php]: what is this one? Can it have an easier to understand name?

Comment (brianteeman) [reply layouts/joomla/form/field/media.php]: guessing from line 106 it should be "modalcontainer" ?

Comment (dgrammatiko) [reply layouts/joomla/form/field/media.php]: Actually my naming convention is bad. I will change all these to hyphen separated words: `basepath` -> `base-path`

Comment (dgrammatiko) [reply layouts/joomla/form/field/media.php]: Ok, attribute names  back to kebab case.
Also fix the stretched preview image 
add (hardcoded ATM) support for more media types
and prepared for the changes in the media manager

**Comment by brianteeman** [comment]
Looks like you have a bug ;)
screenshotr14-43-27

**Comment by dgrammatiko** [comment]
It's this line: `type="image" <? // @TODO add this attribute to the field in order to use it for all media types ?>` should be `type="image" <? /* @TODO add this attribute to the field in order to use it for all media types */ ?>`

File: layouts/joomla/form/field/media.php [hunk 2]
```diff
@@ -76,6 +76,12 @@
 	{
 		$src = JText::_('JLIB_FORM_MEDIA_PREVIEW_EMPTY');
 	}
+
+	if ($showAsTooltip) {
```
Comment (zero-24) [inline layouts/joomla/form/field/media.php]:
```php
	$showPreview = 'static';

	if ($showAsTooltip)
	{
		$showPreview = 'tooltip';
	}
```

Comment (dgrammatiko) [reply layouts/joomla/form/field/media.php]: Re assigning a variable is slower than an if/else but anyways I did it

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on e65c4c1716154b6111e7cd2de9a1fa8cd7d69fc0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17740">issues.joomla.org/tracker/joomla-cms/17740</a>.</sub>

**Comment by Fedik** [comment]
the field seems works, but I cannot see any image in modal, 
maybe because com_media changes

**Comment by dgrammatiko** [comment]
This is already merged, from the new media manager repo

**Pull request closed without merging**